### PR TITLE
Some bytes/str issues in py3 w/ zlib and json

### PR DESCRIPTION
**Pull request by wyndhblb**
zlib wants bytes, json wants strings ...

### CODE REVIEW

**Comment by bkyryliuk** [comment]
@wyndhblb  - could u please add a test to demonstrate the issue?
It would be great to prevent the regression.

**Comment by wyndhblb** [comment]
sure let me refactor things a bit first

**Comment by xrmx** [comment]
I think it would be better that instead of being sloppy on how we call functions and adding wrappers that handles all the the types we just have more care and pass the right types around.

**Comment by wyndhblb** [comment]
sure (python  2+3 makes that hard w/ the type differences) ... six could help, but the default six.b/u does latin-1 and that's not a good encoding default for most modern DBs.

File: tests/utils_tests.py [hunk 1]
```diff
@@ -45,3 +46,9 @@ def test_base_json_conv(self):
     def test_parse_human_timedelta(self, mock_now):
         mock_now.return_value = datetime(2016, 12, 1)
         self.assertEquals(parse_human_timedelta('now'), timedelta(0))
+
+    def test_zlib_compression(self):
+        json_str = """{"test": 1}"""
+        blob = zlib_compress(json_str)
+        got_str = zlib_uncompress_to_string(blob)
+        self.assertEquals(json_str, got_str)
```
Comment (mistercrunch) [inline tests/utils_tests.py]: NIT: missing return char at EOF

**Comment by mistercrunch** [comment]
This LGTM though there's a conflict, please rebase.

[optional] These pure function are nice to test as doctests so that you have an example that is also a test in the docstring. The way things are hooked up it's also accounted for in code coverage analysis.

**Comment by mistercrunch** [comment]
can we keep the formatting things outside this PR, codeclimate doesn't seem to agree with your auto-formatter.

**Comment by rumbin** [comment]
What can be done to get this merged? This bug is effectively keeping me from switching to Python 3 (see #2079), which I would love to do, as Python 2 has its encoding issues of its own that I'm struggling with.

**Comment by mistercrunch** [comment]
This PR needs to:
* rebase 
* remove all of the unnecessary formatting/indent changes
* pass the build

**Comment by rumbin** [comment]
See alternative PR #2558, as this one doesn't seem to move on.

**Comment by mistercrunch** [comment]
superseeded

**Pull request closed without merging**